### PR TITLE
[Internal] Code maintenance: Add code cleanup and refactor

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description

1. Moved editorconfig to the solution so it get's applied to all projects.
2. Applied pattern matching to most files. Excluded v2 forked files.
3. Moved all feed iterators into their own folder.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs [hunk 1]
```diff
@@ -210,7 +210,7 @@ internal override void PopulateRequestOptions(RequestMessage request)
             }
             else
             {
-                request.Headers.Add(HttpConstants.HttpHeaders.ContentSerializationFormat, ContentSerializationFormat.CosmosBinary.ToString());
+                //request.Headers.Add(HttpConstants.HttpHeaders.ContentSerializationFormat, ContentSerializationFormat.CosmosBinary.ToString());
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs:213]: Isn't it an independent PR?

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs:213]: Waiting on it.

Comment (j82w) [reply Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs]: I was testing this on a different branch and forgot to stash the change.

File: Microsoft.Azure.Cosmos/src/Batch/BatchAsyncContainerExecutor.cs [hunk 2]
```diff
@@ -163,9 +163,8 @@ private static IDocumentClientRetryPolicy GetRetryPolicy(RetryOptions retryOptio
                             | itemRequestOptions.Properties.TryGetValue(HttpConstants.HttpHeaders.PartitionKey, out object pkStringObj)))
             {
                 byte[] epk = epkObj as byte[];
-                string epkStr = epkStrObj as string;
                 string pkString = pkStringObj as string;
-                if ((epk == null && pkString == null) || epkStr == null)
+                if ((epk == null && pkString == null) || !(epkStrObj is string epkStr))
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Batch/BatchAsyncContainerExecutor.cs:167]: nit:
```suggestion
                if ((epk == null && pkString == null) || !(epkStrObj is string _))
```

Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Batch/BatchAsyncContainerExecutor.cs:167]: you can also do `epk is null` instead in the spirit of pattern matching.

Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Batch/BatchAsyncContainerExecutor.cs:167]: I applied the formatting fix via VS to the entire project. It provides a list of all the files and changes it does. I manually went through and unchecked files I knew are v2 forks like document client.

**Pull request merged**